lib/util/utils.ts

```diff
@@ -84,15 +84,38 @@ export type ParsedOutput<T> = {
 
 /**
  * Parses the connection string and returns an object of type T.
+ *
+ * Connection strings have the following syntax:
+ *
+ * ConnectionString ::= Part { ";" Part } [ ";" ] [ WhiteSpace ]
+ * Part             ::= [ PartLiteral [ "=" PartLiteral ] ]
+ * PartLiteral      ::= [ WhiteSpace ] Literal [ WhiteSpace ]
+ * Literal          ::= ? any sequence of characters except ; or = or WhiteSpace ?
+ * WhiteSpace       ::= ? all whitespace characters including \r and \n ?
+ *
  * @param {string} connectionString The connection string to be parsed.
  * @returns {ParsedOutput<T>} ParsedOutput<T>.
  */
 export function parseConnectionString<T>(connectionString: string): ParsedOutput<T> {
-  return connectionString.split(';').reduce((acc, part) => {
+  return connectionString.trim().split(';').reduce((acc, part) => {
+    part = part.trim();
     const splitIndex = part.indexOf('=');
+
+    if (part === '') {
```

Comment:
I would move this above line 102. That way you don't end up doing the call to indexof if `part` is empty

Reply:
Good call, looks very nice that way.